Sources/LiveViewNative/Utils/WrappedShapeStyle.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  WrappedShapeStyle.swift
+// LiveViewNative
+//
+//  Created by May Matyi on 2/17/23.
+//
+
+import SwiftUI
+
+struct WrappedShapeStyle: Decodable {
```

Comment:
This may be better represented as an `enum`:

```swift
enum WrappedShapeStyle: Decodable {
  case color(Color)
  case linearGradient(LinearGradient)

  ...
}
```

It could even be an extension on [`AnyShapeStyle`](https://developer.apple.com/documentation/swiftui/anyshapestyle):

```swift
extension AnyShapeStyle: Decodable {
  init(from decoder: Decoder) throws {
    ...
    // if it's a color
    self = Self(color)
    // if it's a gradient
    self = Self(gradient)
  }
}
```
If you go with an extension, there would be no need to switch over the `concreteStyle` in the modifiers, you can just use the decoded style directly.

Reply:
Thanks, adding an extension for `AnyShapeStyle` seems to be the way to go! Fixed 🙂